src/main/webapp/search/SearchResults.js

```diff
@@ -1,136 +1,231 @@
-import React from 'react';
-import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
-import ExportExcel from './ExportExcel'
+import React, { Component } from "react";
+import { BootstrapTable, TableHeaderColumn } from "react-bootstrap-table";
+import ExportExcel from "./ExportExcel";
+import { User } from "../util/ajax";
 
-function linkFormatter(cell, row) {
-    return '<a href="' + row.link + '">' + row.key + '</a>';
-}
-
-let excelData = [];
-let formattedProjectData = [];
+class SearchResults extends Component {
+  excelData = [];
+  formattedProjectData = [];
 
-const excelDataSet = [
+  excelDataSet = [
     {
-        columns: ["Project Key", "Title", "Project Type", "Status", "Update Date", "Expiration Date"],
-        data: excelData
+      columns: [
+        "Project Key",
+        "Title",
+        "Project Type",
+        "Status",
+        "Update Date",
+        "Expiration Date"
+      ],
+      data: this.excelData
     }
-];
+  ];
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: {
+        displayName: "",
+        userName: "",
+        emailAddress: ""
+      }
+    };
+  }
 
-function compareDateString(a, b, order) {
+  componentDidMount() {
+    User.getUserSession(this.props.getUserUrl).then(resp =>
+      this.setState({ user: resp.data })
+    );
+  }
 
+  linkFormatter = (cell, row) => {
+    if (row.linkDisabled === false) {
+      return '<a href="' + row.link + '">' + row.key + "</a>";
+    } else {
+      return row.key;
+    }
+  };
+
+  compareDateString = (a, b, order) => {
     // This format comes from the server - update if the format changes.
     const format = "MM/DD/YYYY";
     let aTime = moment(a, format);
     let bTime = moment(b, format);
-    let aIsNewer = (aTime.isValid() && bTime.isValid() && aTime > bTime);
-    let bothSame = (aTime.isValid() && bTime.isValid() && aTime === bTime);
+    let aIsNewer = aTime.isValid() && bTime.isValid() && aTime > bTime;
+    let bothSame = aTime.isValid() && bTime.isValid() && aTime === bTime;
     if (!aTime.isValid() && bTime.isValid()) {
-        aIsNewer = false
+      aIsNewer = false;
     }
     if (!bTime.isValid() && aTime.isValid()) {
-        aIsNewer = true
+      aIsNewer = true;
     }
     if (!aTime.isValid() && !bTime.isValid()) {
-        bothSame = true
+      bothSame = true;
     }
-    if (order === 'desc') {
-        if (aIsNewer) return -1;
-        if (bothSame) return 0;
-        return 1;
+    if (order === "desc") {
+      if (aIsNewer) return -1;
+      if (bothSame) return 0;
+      return 1;
     } else {
-        if (aIsNewer) return 1;
-        if (bothSame) return 0;
-        return -1;
+      if (aIsNewer) return 1;
+      if (bothSame) return 0;
+      return -1;
     }
-}
+  };
 
-function expirationSort(a, b, order) {
+  expirationSort = (a, b, order) => {
     return compareDateString(a.expiration, b.expiration, order);
```

Comment:
Since this is no longer a function call, should this be `this. compareDateString` ? Both here and below.